### ISSUE TITLE
Maybe a bug on Adding an Entry

### ISSUE THREAD

**Issue by sbservices** [alicolville/meal-tracker#167]
This may or may not be a bug.

I am just in our demo area logged in as Maeve Hoban (made up user!), I went to add a new food item, I clicked on the Cheese (which you added as a demo food to test) and I clicked on 'Add' instead of 'Add and Return'.  Then I clicked on 'Crackers' and clicked on 'Add and Return' and it asked me for a quantity so I selected 3.  It didn't ask me for how many portions of cheese it just proceeded to add it in the background I can only assume, and now when I check the back office, Cheese is in there twice. 

<img width="983" alt="image" src="https://github.com/alicolville/meal-tracker/assets/101115016/fa2b15b3-5dd6-4df2-b33a-c2e1bfc8636c">

And as an End user, I see 'Cheese' twice and if I log in as a different user, I logged in as 'Mary Kelly'. I shouldn't be able to see Maeve's cheese!! (Sounds very wrong) 😂. 

![WhatsApp Image 2023-08-24 at 17 22 31](https://github.com/alicolville/meal-tracker/assets/101115016/3e9c2fa1-f6ec-44df-b4f3-13dba5a8ab2b)

And I have 'see other users entries' set to 'no'.
<img width="463" alt="image" src="https://github.com/alicolville/meal-tracker/assets/101115016/7a84d04c-9878-4725-aec6-eec33ed9c133">

So could there be a glitch that whereby clicking on 'add' rather than 'add and return' is creating a new meal entry with admin flag set to yes?

See here in database screenshot - Maeve Hoban is user #11 (you are user ID #4) and User #11's permissions is 'Customer', not 'Administrator'.  But it has the administrator flag marker set to #1 . 

<img width="860" alt="image" src="https://github.com/alicolville/meal-tracker/assets/101115016/d6eae7e3-9f9e-4845-aee8-b7c23774fb51">

**Comment by sbservices** [comment]
It seems to be something to do with when you have 'connect to external sources' enabled because you get a new icon appear to allow you to search using the magnifying glass.  Then when you've searched (the external data source) the entry gets added with admin flag.  We observed this in WeighlessWonders system and we turned off the external searching feature. But we didn't know how it was happening, but now I've found the trigger.

**Comment by alicolville** [comment]
Yeah, it shouldn't set the "added_by_admin" to 1 when it's fetched the meal from an external source. I'll fix that now!

**Comment by alicolville** [comment]
Fix in: https://github.com/alicolville/meal-tracker/pull/168

**Comment by alicolville** [comment]
Fixed in 3.1.2 (https://github.com/alicolville/meal-tracker/pull/168).

Thanks for your help @sbservices !
